### PR TITLE
Clean code

### PR DESCRIPTION
**Pull request by guizmaii**

### CODE REVIEW

**Review by sderosiaux — APPROVED** [review]

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -867,39 +878,45 @@ object AdminClient extends Accessible[AdminClient] {
...
           .asScala
           .toList
-      ) { jNode =>
-        ZIO.getOrFailWith(new RuntimeException("NoNode node not expected among topic replicas"))(Node(jNode))
-      }
+          .forEach { jNode =>
+            Node(jNode) match {
+              case Some(node) => Success(node)
+              case None       => Failure(new RuntimeException("NoNode node not expected among topic replicas"))
+            }
```
Comment (strokyl) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:891]: :green_circle: 
In what this is better than previous `ZIO.getOrFailWith`?

Comment (guizmaii) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:891]: You should not use ZIO for pure code. ZIO means that there's an effect. There's none here. It's a pure transformation.

**Comment by strokyl** [comment]
:red_circle: 
I don't understand the purpose of this PR?
Do you plan to merge it upstream? Because otherwise we just have `cdk-master` diverging from upstream master and I don't think we want that to happen at all?

**Comment by guizmaii** [comment]
@strokyl 

> I don't understand the purpose of this PR?

Clean the code. Remove inappropriate ZIO usages. 

> Do you plan to merge it upstream? 

Of course

**Pull request merged**